### DISCUSSION TITLE
Terminal size too small

### DISCUSSION THREAD

**Discussion by nalzok** [ghostty-org/ghostty#4104] [Feature Requests, Ideas] [answered]
Running [btop](https://github.com/aristocratos/btop) with the default configuration gives the following error. Of course, a user can resize the terminal to make it work, but I guess we need a more sensible default for `window-width` and `window-height`?

(PS: as the screenshot illustrates, there is something wrong with ANSI escape sequence rendering, but that is a separate issue)

<img width="400" alt="image" src="https://github.com/user-attachments/assets/2976025e-3911-4dc7-bd89-a222b13cb868" />

**Comment by mitchellh** [comment, marked as answer]
Our default size is based on a pixel size which means font size impacts it. I think the default is fine.

**Reply by nalzok** [reply]
I beg to differ. It's basically a de facto standard that the terminal size should be at least 24 by 80, and violating this assumption makes many applications unusable (e.g., `btop`). To see if your sizing algorithm is really a sensible default, I'll keep using 79 by 27 for now, and see if having a small terminal causes other problems.

**Reply by mitchellh** [reply]
I expect most people immediately resize the window to some degree esp if using more power tools such as an editor or resource monitoring.

**Reply by JakeWharton** [reply]
I would also like to advocate for 80xSomething exactly. Even if you customize the padding and decorations, so long as the window size isn't explicitly set, being sized to perfectly fit an exact and standard cell size is nice.

This extends naturally from the discussion about [quick terminal initial size offering cell-based sizing](https://github.com/ghostty-org/ghostty/issues/2384#issuecomment-2401129283), and the idea that [window decorations/padding should not break the snap resizing](https://github.com/ghostty-org/ghostty/discussions/2946).

If those two things were supported, doing this would effectively be equivalent to the user specifying a window size of 24r,80c (or whatever the syntax would be). That would become the default window size, and the machinery for exact sizing cells regardless of window decor would do the rest.
